### PR TITLE
Style improvements for template previews

### PR DESCRIPTION
**Pull request by koke**
## Description

This PR addresses the feedback in https://github.com/wordpress-mobile/gutenberg-mobile/issues/1777, see the discussion on that issue for more details on the changes.

In summary, I made the new `ModalHeaderBar` look more like the system one and added dark mode support.

gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1785

## Screenshots 

spt-style-fixes

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/editor/src/store/actions.native.js [hunk 1]
```diff
@@ -25,3 +30,12 @@ export function togglePostTitleSelection( isSelected = true ) {
 export function* autosave() {
 	RNReactNativeGutenbergBridge.editorDidAutosave();
 }
+
+/**
+ * Action generator to set the selected layout template.
+ *
+ * @param {Object} template Layout's template
+ */
+export function* setLayoutTemplate( template ) {
```
Comment (koke) [inline packages/editor/src/store/actions.native.js]: @geriux I originally created my branch from yours but merged master after yours was merged 😅 Is this expected to be here or is it a bad merge?

Comment (geriux) [reply packages/editor/src/store/actions.native.js]: Oh no, this isn't supposed to be here =(

Comment (koke) [reply packages/editor/src/store/actions.native.js]: 👍 Reverted those changes in 8af053d8ff

**Review by geriux — APPROVED** [review]
LGTM! Good work with the styles and alignment of the Header content 👏🎉

Tested both on iOS (iPhone / iPad) with / without darkmod and Android.

**Pull request merged**